src/inmanta/parser/plyInmantaParser.py

```diff
@@ -1019,10 +1030,21 @@ def convert_to_references(variables: List[Tuple[str, LocatableString]]) -> List[
                 var_parts[1:], lambda acc, part: acc + len(part) + 1, initial=end_char + 1
             )
             for attr, char_offset in zip(var_parts[1:], attribute_offsets):
+                field_name_left_trim = attr.lstrip()
+                left_spaces = len(attr) - len(field_name_left_trim)
+                field_name_full_trim = field_name_left_trim.rstrip()
+                right_spaces = len(field_name_left_trim) - len(field_name_full_trim)
+
                 range_attr: Range = Range(
-                    var.location.file, var.location.lnr, char_offset, var.location.lnr, char_offset + len(attr)
+                    var.location.file,
+                    var.location.lnr,
+                    char_offset + left_spaces,
+                    var.location.lnr,
+                    char_offset + len(attr) - right_spaces,
+                )
+                attr_locatable_string: LocatableString = LocatableString(
+                    field_name_full_trim, range_attr, var.lexpos, var.namespace
                 )
```

Comment:
I think this method could become a whole lot easier to read if you extract this part into a function with a clear name.